subsys/usb/device/Kconfig

```
@@ -48,6 +48,23 @@ config USB_DEVICE_SN
 	  Serial Number String will be derived from
 	  Hardware Information Driver (HWINFO).
 
+config USB_DEVICE_CONFIGURATION_ENABLE
+	bool "USB configuration descriptor support"
+	default n
+	help
+	  Select 'y' to make the USB configuration descriptor available. The
+	  configuration descriptor is not available by default.
```

Both "CONFIGURATION_ENABLE" and "configuration descriptor" are too confusing, there is always a configuration descriptor. What you want to enable here is a additional string descriptor that describes this single configuration.

```suggestion
config USB_CONFIGURATION_STRING_DESC_ENABLE
	bool "USB configuration string descriptor support"
	help
	  Enable string descriptor that describes the default configuration.
	  Since there is only one configuration, this string descriptor may not
	  provide much useful information.
```

Also, `default n` and `Select 'y' to make` are redundant.

---

Fixed.